freelist.go

```diff
@@ -326,3 +327,44 @@ func (f *freelist) reindex() {
 		}
 	}
 }
+
+func (f *freelist) addReadonlyTXID(tid common.Txid) {
+	f.readonlyTXIDs = append(f.readonlyTXIDs, tid)
+}
+
+func (f *freelist) removeReadonlyTXID(tid common.Txid) {
+	for i := range f.readonlyTXIDs {
+		if f.readonlyTXIDs[i] == tid {
+			last := len(f.readonlyTXIDs) - 1
+			f.readonlyTXIDs[i] = f.readonlyTXIDs[last]
+			f.readonlyTXIDs = f.readonlyTXIDs[:last]
+			break
+		}
+	}
+}
+
+type txIDx []common.Txid
+
+func (t txIDx) Len() int           { return len(t) }
+func (t txIDx) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
+func (t txIDx) Less(i, j int) bool { return t[i] < t[j] }
+
+// freePages releases any pages associated with closed read-only transactions.
+func (f *freelist) freePages() {
+	// Free all pending pages prior to earliest open transaction.
+	sort.Sort(txIDx(f.readonlyTXIDs))
+	minid := common.Txid(0xFFFFFFFFFFFFFFFF)
```

Comment:
while we're at it, let's also make this `math.MaxUint64` - slightly less confusing to read

Reply:
Yes, updated & thx